OpenRA.Mods.Common/Pathfinder/PathGraph.cs

```diff
@@ -160,21 +186,32 @@ public List<GraphConnection> GetConnections(CPos position)
 				{
 					var layerPosition = new CPos(position.X, position.Y, cli.Layer.Index);
 					var entryCost = cli.Layer.EntryMovementCost(Actor.Info, locomotor.Info, layerPosition);
-					if (entryCost != CostForInvalidCell)
+					if (entryCost != CostForInvalidCell &&
+						CanEnterNode(layerPosition) &&
+						this[layerPosition].Status != CellStatus.Closed)
```

Comment:
Wouldn't it be 'faster' to swap these two lines? First check if `Closed`?

Reply:
The movement cost check must occur first as it validates the cell is within bounds - I don't believe we can safely index without that check.

Reply:
Btw, if we're already passing `layerPosition` to `CanEnterNode`, would it make sense to move the `this[layerPosition].Status != CellStatus.Closed` check inside that function as well?

Reply:
I like the symmetry with the ground layer check above where the closed check is in the condition, but I guess I'm not too tied either way:
`if (movementCost != CostForInvalidCell && info[neighbor].Status != CellStatus.Closed)`